### PR TITLE
[13564] account-avatar component

### LINKED ISSUE #13564: UI Component - Account Avatar
This issue is to track the creation of the component: **Account Avatar**

You can access the link at https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Mobile-Components?node-id=483%3A19401.

Please have a chat with the design team before starting with the component, and don't forget to add sizing to the issue

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #13564 
status: ready

### CODE REVIEW

File: src/quo2/components/account_avatar.cljs
Comment (flexsurfer) [inline src/quo2/components/account_avatar.cljs]: just wondering why `background-color` not in style map

File: src/quo2/components/account_avatar.cljs
Comment (flexsurfer) [inline src/quo2/components/account_avatar.cljs]: could elaborate on why peach icon ?

**Review by flexsurfer — COMMENTED** [review]
great job

File: src/quo2/components/account_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Because it's derived from if the style is `light` or `dark`

File: src/quo2/components/account_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: That's a mistake, I should make it configurable by choosing an icon
Will do it today

File: src/quo2/components/account_avatar.cljs
Comment (flexsurfer) [reply src/quo2/components/account_avatar.cljs]: im confused even more now :)

Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Sorry for this confusion 🤣 
I will take care of the comments on this PR

Comment (flexsurfer) [reply src/quo2/components/account_avatar.cljs]: ok so my question
`:background-color` prop should be inside :style map, view doesn't have `:background-color` prop i guess ? or am i wrong ?

Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: No you're totally correct
I'll look into it shortly

Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Styling should be inside a :style map inside the prop map

File: src/quo2/components/account_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Done

File: src/quo2/components/account_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Done

File: src/quo2/components/account_avatar.cljs
Comment (flexsurfer) [inline src/quo2/components/account_avatar.cljs]: could you elaborate on why icon is a string? 
also `main-icons` ns is used only for IDE, it isn't used in icon component,

Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Hey @flexsurfer
I wanted to create a namespaced keyword to make the icon dynamic

Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: What do you mean by main-icons is only for IDE?

Comment (flexsurfer) [reply src/quo2/components/account_avatar.cljs]: i mean its just for find function in IDE, when you want to find keyword, to find icon keyword this ns is used, but in icon component we remove this ns,

Comment (flexsurfer) [reply src/quo2/components/account_avatar.cljs]: > I wanted to create a namespaced keyword to make the icon dynamic

what do you mean ?

Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Oh i see now, but the usages I see everywhere has this form

`:main-icons/wallet`

And namespaced keywords help show ownership of a keyword by namespacing it


If we have 2 id keywords they would shadow each other on usage

`:user/id`
`product/id`

But with namespaced keywords, each is tied to a certain namespace

Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: > i mean its just for find function in IDE, when you want to find keyword, to find icon keyword this ns is used, but in icon component we remove this ns,

So you mean i should do it like this
`[Icons/icon :peach16 {props}]` ?

Comment (flexsurfer) [reply src/quo2/components/account_avatar.cljs]: yeah i mean it should be keyword not string `[icons/icon icon]`

Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Oh I already converted the string to a keyword here using the `keyword` function 
I used `str` to make use of the string coming from the text field and form a namespaced keyword of
`:main-icons/<DYNAMIC-ICON>`

Comment (flexsurfer) [reply src/quo2/components/account_avatar.cljs]: if `icon` is already a keyword, then all is good just use `[icons/icon icon]`

Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Yeah I'd do that
But it isn't, it's passed as a string prop

Comment (flexsurfer) [reply src/quo2/components/account_avatar.cljs]: it should be a keyword then

Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: If we passed it as a keyword, We'd have to make a list of keywords for all possible emojis .. Which one would you prefer ? converting the string to a keyword or make a keyword only list of emojis ?

Comment (flexsurfer) [reply src/quo2/components/account_avatar.cljs]: keyword only list of emojis

File: src/quo2/components/account_avatar.cljs
Comment (Parveshdhull) [inline src/quo2/components/account_avatar.cljs]: (same here) better to confirm with design team, but it looks like this component uses emoji and constructs icon in component

Comment (Parveshdhull) [reply src/quo2/components/account_avatar.cljs]: also, `dark?` (use app theme)

File: src/quo2/components/account_avatar.cljs
Comment (Parveshdhull) [inline src/quo2/components/account_avatar.cljs]: ?

File: src/quo2/screens/account_avatar.cljs
Comment (Parveshdhull) [inline src/quo2/screens/account_avatar.cljs]: Please, assign default state of all keys

**Comment by siddarthkay** [comment]
@cammellos @ibrkhalil : I'll be taking this up for a quick Design Review

**Comment by siddarthkay** [comment]
@ibrkhalil  Design issue 1 

There seems to be some issue with the dimensions of the icon that you have used, in this scenario it gets cut from the right hand side. you also need to work on retrieving the values of dark/light modes from theme instead of adding a boolean input field.

Screenshot 2022-08-22 at 4 08 26 PM

**Comment by siddarthkay** [comment]
Looking at assets you have imported in your PR it seems that you have imported the 15px version instead of the 20px version, that should take care of design issue 1 

Screenshot 2022-08-22 at 4 14 35 PM

File: src/quo2/components/account_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Done

File: src/quo2/components/account_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Fixed

File: src/quo2/screens/account_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/screens/account_avatar.cljs]: Done

**Comment by ibrkhalil** [comment]
> Looking at assets you have imported in your PR it seems that you have imported the 15px version instead of the 20px version, that should take care of design issue 1
> 
> Screenshot 2022-08-22 at 4 14 35 PM

Done, Thanks Sid

**Comment by churik** [comment]
@ibrkhalil please, re-request review on this PR and rebase your work to current `develop`, I'll re-run failed tests and it is ready to merge if you're sure it couldn't cause regression in the existing app.

File: src/quo2/components/account_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Done

Comment (ibrkhalil) [reply src/quo2/components/account_avatar.cljs]: Thanks Parvesh .. I'll ask them before merging

File: src/quo2/components/avatars/account_avatar.cljs
Comment (flexsurfer) [inline src/quo2/components/avatars/account_avatar.cljs]: is there default size ?

File: src/quo2/components/avatars/account_avatar.cljs
Comment (flexsurfer) [inline src/quo2/components/avatars/account_avatar.cljs]: better to change places `(sizes size)` or even better `(get sizes size)`

**Review by flexsurfer — APPROVED** [review]

File: src/quo2/components/avatars/account_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/avatars/account_avatar.cljs]: Done

File: src/quo2/components/avatars/account_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/avatars/account_avatar.cljs]: Added defaults

File: src/quo2/components/avatars/account_avatar.cljs
Comment (Parveshdhull) [inline src/quo2/components/avatars/account_avatar.cljs]: Please don't assign names, only use size literals like 80, 48 and calculate other params (border-radii, inner-icon-sizes) based on that

File: src/quo2/components/avatars/account_avatar.cljs
Comment (flexsurfer) [inline src/quo2/components/avatars/account_avatar.cljs]: my comment was not only about sizes but for all maps avatar-border-radius and inner-icon-size as well

Comment (ibrkhalil) [reply src/quo2/components/avatars/account_avatar.cljs]: Okay, Doing now

File: src/quo2/components/avatars/account_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/avatars/account_avatar.cljs]: Alright then .. On it

Comment (ibrkhalil) [reply src/quo2/components/avatars/account_avatar.cljs]: Can you please check now Parvesh ?

File: src/quo2/components/avatars/account_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/avatars/account_avatar.cljs]: Done, Applied Parvesh's refactor ..

**Review by Parveshdhull — APPROVED** [review]

File: src/quo2/components/avatars/account_avatar.cljs
Comment (Parveshdhull) [reply src/quo2/components/avatars/account_avatar.cljs]: looks good, thank you

File: src/quo/design_system/colors.cljs
Comment (Parveshdhull) [inline src/quo/design_system/colors.cljs]: Hi, why changing `quo/design_system/colors`? (this will also affect old ui)
And we should only use `quo2.colors` for new ui.

File: src/quo2/components/avatars/user_avatar.cljs
Comment (Parveshdhull) [inline src/quo2/components/avatars/user_avatar.cljs]: Why changes in user avatar?

**Review by Parveshdhull — CHANGES_REQUESTED** [review]

File: src/quo/design_system/colors.cljs
Comment (ibrkhalil) [reply src/quo/design_system/colors.cljs]: For [this ](https://discord.com/channels/624205794384281629/626744384565674034/1016761652080824453)request

File: src/quo2/components/avatars/user_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/avatars/user_avatar.cljs]: I added this fix because when the team updated the colors ns, The color for the component was different.

File: src/quo/design_system/colors.cljs
Comment (Parveshdhull) [reply src/quo/design_system/colors.cljs]: > For this request
> @Mobile any chance we can get the default background of the testing area to be neutral 90?

We should not change color's value, we should change which color to use for preview background

Comment (ibrkhalil) [reply src/quo/design_system/colors.cljs]: Correct, Thanks Parvesh!
I'll do it now

Comment (ibrkhalil) [reply src/quo/design_system/colors.cljs]: Done

File: src/quo2/components/avatars/user_avatar.cljs
Comment (ibrkhalil) [reply src/quo2/components/avatars/user_avatar.cljs]: Apparently this issue isn't happening on latest dev
Reverted

File: src/quo/design_system/colors.cljs
Comment (Parveshdhull) [reply src/quo/design_system/colors.cljs]: > Done

[src/quo/design_system/colors.cljs](https://github.com/status-im/status-mobile/pull/13700/files#diff-034ed1cf36e210d1c972dcd619946509eb41afefec8b8d069d0dadbc84d60aba) still has changes, please revert it.

**Comment by Parveshdhull** [comment]
Hi @ibrkhalil, Please can you rebase this PR and remove unused changes so that we can merge it?

File: src/quo2/screens/avatars/account_avatar.cljs
Comment (Parveshdhull) [inline src/quo2/screens/avatars/account_avatar.cljs:29]: I am not sure how these options will be informative to the tester/reviewer. But still thank you for removing unused changes

**Review by Parveshdhull — APPROVED** [review]

**Pull request merged**
